.github/actions/build-image/action.yaml

```diff
@@ -0,0 +1,177 @@
+---
+name: build-image
+description: |
+  The action is launched in the PRs opened by the Renovate bot
+  when one or more of the tags `build:dev`, `build:staging`, `build:prod` are added,
+  and does the following:
+    - export environment variables from required files
+    - extract tag by image digest specified in Dockerfile
+    - build and publish image with received tag and `latest` tag
+    - create a json file with variables for the jinja template and upload it as an artifact
+    - send a notification to slack channel
+
+inputs:
+  context:
+    description: folder with files required to build docker image
+    required: true
+  dockerhub_token:
+    description: token for read-only access to dockerhub
+    required: false
+    default: ''
+  dockerhub_username:
+    description: username for read-only access to dockerhub
+    required: false
+    default: ''
+  environment:
+    description: |
+      the environment for which the image will be built:
+      `dev`, `prod`, `staging`
+    required: true
+  slack_webhook_url:
+    description: |
+      CI slack channel webhook url for sending build notifications
+      https://api.slack.com/apps/A01LM626QTZ/incoming-webhooks
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Combine all .env files into .env.common file
+      id: env-files
+      uses: ftnext/action-concatenate-files@v1.0.1
+      with:
+        sources: |
+          .env
+          .env.${{ inputs.environment }}
+          ${{ inputs.context }}/.env
+          ${{ inputs.context }}/.env.${{ inputs.environment }}
+        destination: .env.common
+    - name: Export environment variables from .env.common file
+      id: env-common
+      uses: c-py/action-dotenv-to-setenv@v5
+      with:
+        env-file: .env.common
+    - name: Configure AWS credentials
+      id: aws-credentials
+      uses: aws-actions/configure-aws-credentials@v4.3.1
+      with:
+        aws-region: ${{ env.aws_region }}
```

Comment:
I think this should come from inputs. Currently this GH action says nothing that it needs any AWS access, but it does. That shouldn't be like that